### PR TITLE
fix/allowed_options

### LINKED ISSUE #4536: v9_gui: admin role issue.
Let's create a admin role hotel:
image

with the following user options:
image

If i assign a local user to this admin role, connect to the admin gui with that user and i try to create a user then the access duration list is empty and the role list contain all the role available.

### LINKED ISSUE #4854: Admin roles: Allowed node roles not taken into account
When a user has an admin role with a limited list of `Allowed node roles`, he is able to create nodes with all roles (not only `Allowed node roles`).

**To Reproduce**
1. Create two roles: `role1` and `role2`
1. Create an admin role called `Role-1`
   * Actions: Nodes - Create
   * Node options - Allowed node roles: `role1`
1. Create a user `foo` and assign `Role-1` as access level
1. Log in as `foo` user and try to create a node with `role2`

=> Node is created.

**Expected behavior**
User `foo` should be able to create only nodes with role `role1`.

### PR DESCRIPTION
**Pull request by jrouzierinverse**
Description
===========
Use the Admin Roles allowed_access_levels

Issue
=====
Fixes #4536
Fixes #4854

Checklist
=========
- [ ] Document the feature
- [x] Add unit tests
- [ ] Add acceptance tests (TestLink)

### CODE REVIEW

**Comment by fdurand** [comment]
Still missing the access duration.
Also the user need to have the "Nodes - Read" action to be able to see the roles.
Add all the roles appear in the list and not only the one allowed in the admin role.

**Comment by julsemaan** [comment]
@jrouzierinverse 

Can you fix this ASAP so that it can get into 9.0.1

**Comment by julsemaan** [comment]
Pushing to 9.0.2 so we can get a full fix for the issue

**Comment by nqb** [comment]
We will need this PR merged for 9.2.

**Comment by julsemaan** [comment]
@jrouzierinverse 

Could you work on a full fix based on @fdurand's comments. Its indeed overdue as the current admin lacks some advanced access control due to that missing feature

**Comment by extrafu** [comment]
@jrouzierinverse 

Make sure we handle roles and access duration validation.

**Comment by extrafu** [comment]
@julsemaan If all good please merge.

**Comment by julsemaan** [comment]
Given @fdurand is the assignee and probably knows the problem with more details than me I'll just let him double check the PR to ensure it meets what he expects for the fix

**Comment by fdurand** [comment]
Still doesn't work, from a user i created with node_manager role, i am able to change the role of a device to any role and same for user and role/access duration.
Also my own user is able to change his own admin role and grant access to ALL.

**Comment by jrouzierinverse** [comment]
Let me see

**Comment by fdurand** [comment]
Nodes options looks good.
User options no.
I am not able to set any role/access duration/access level

**Comment by fdurand** [comment]
After added:
Security event  - READ
Nodes - READ
Configuration Main - READ
System - READ

I am able to set the role of the device (only the allowed roles) but i can set any access duration.

File: html/pfappserver/root/static.alt/src/globals/configuration/pfConfiguration.js [hunk 1]
```diff
@@ -514,6 +543,38 @@ export const pfConfigurationActions = {
       }
     }
   },
+  set_role_by_acl_node: {
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/globals/configuration/pfConfiguration.js]: @satkunas This action doesn't seem to be used anywhere. Forgot to commit your changes to the Node components?

File: html/pfappserver/root/static.alt/src/components/pfCSVImport.vue [hunk 2]
```diff
@@ -741,17 +686,33 @@ export default {
...
-        const field = this.fields[index]
-        if ('types' in field && field.types.includes(type)) {
-          return true
+    isComponentType (componentTypes, { key }) {
+      if (key) {
+        const index = this.fields.findIndex(field => field.value === key)
+        if (index >= 0) {
+          const field = this.fields[index]
+          for (let t = 0; t < componentTypes.length; t++) {
+            if (field.types.map(type => fieldTypeComponent[type]).includes(componentTypes[t])) return true
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/components/pfCSVImport.vue]: @satkunas Put the `map` outside the *for* loop for better performance.

**Review by cgx — CHANGES_REQUESTED** [review]

**Review by cgx — APPROVED** [review]

**Pull request merged**